### PR TITLE
feat: navigation-menu

### PR DESCRIPTION
**Pull request by Ventinc**
# Navigation Menu Primitive

https://user-images.githubusercontent.com/16818589/215111651-a38014c2-6177-4c29-9f9b-84223cb0c46e.mov

### CODE REVIEW

**Comment by shadcn** [comment]
What!! Thank you. Let me QA.

File: apps/www/components/ui/navigation-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      "group flex flex-1 list-none justify-center rounded-lg border border-slate-300 bg-white p-1 dark:border-slate-700 dark:bg-slate-800",
+      className
+    )}
+    {...props}
+  />
+)
+
+export const NavigationMenuItem = NavigationMenuPrimitive.NavigationMenuItem
+
+const navigationMenuTriggerStyle = cva(
```
Comment (shadcn) [inline apps/www/components/ui/navigation-menu.tsx:31]: Can we inline the classes in the `NavigationMenuTrigger`? Are we re-using this elsewhere?

File: apps/www/components/ui/navigation-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  "rounded-md py-2 px-3 font-medium leading-none hover:bg-slate-100 focus:shadow-sm focus:ring-2 focus:ring-slate-300 focus:bg-slate-100 dark:focus:bg-slate-700 dark:hover:bg-slate-700 dark:focus:ring-slate-500 outline-none w-full transition-all duration-300"
+)
+
+export const NavigationMenuTrigger: React.FC<
```
Comment (shadcn) [inline apps/www/components/ui/navigation-menu.tsx:35]: We need to add the `displayName` for the components.

`NavigationMenuTrigger.displayName = NavigationMenuPrimitive.Trigger.displayName`

File: apps/www/components/ui/navigation-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  >
+    {children}{" "}
+    <ChevronDown
+      className="relative top-[1px] h-4 w-4 transition duration-200 group-data-[state=open]:rotate-180"
+      aria-hidden
+    />
+  </NavigationMenuPrimitive.Trigger>
+)
+
+const navigationMenuLinkVariants = cva([], {
```
Comment (shadcn) [inline apps/www/components/ui/navigation-menu.tsx:54]: Yeah. We can skip variants and keep the styles inline. The consumers of the component can adjust the styles as needed.

File: apps/www/components/ui/navigation-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      className={cn(
+        "origin-top-center relative mt-1.5 h-[var(--radix-navigation-menu-viewport-height)] w-full overflow-hidden rounded-md border border-slate-200 bg-white shadow-lg data-[state=open]:animate-in data-[state=open]:zoom-in dark:border-slate-700 dark:bg-slate-800 md:w-[var(--radix-navigation-menu-viewport-width)]",
+        className
+      )}
+      {...props}
+    />
+  </div>
+)
+
+export const NavigationMenuSub = NavigationMenuPrimitive.Sub
```
Comment (shadcn) [inline apps/www/components/ui/navigation-menu.tsx:122]: Let's also move all export at the end of the file. See

https://github.com/shadcn/ui/blob/6ea6856f5a1082d4d9c231559b6bc3ee73827493/apps/www/components/ui/dropdown-menu.tsx#L187-L203

I like inline export too but I found it a bit easier to manage when we have all export grouped together in the case of primitives.

File: apps/www/components/ui/navigation-menu.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    {...props}
+  />
+)
+
+export const NavigationMenuList: React.FC<
+  NavigationMenuPrimitive.NavigationMenuListProps
+> = ({ className, ...props }) => (
+  <NavigationMenuPrimitive.List
+    className={cn(
+      "group flex flex-1 list-none justify-center rounded-lg border border-slate-300 bg-white p-1 dark:border-slate-700 dark:bg-slate-800",
```
Comment (shadcn) [inline apps/www/components/ui/navigation-menu.tsx:22]: Swap `rounded-lg` with `rounded-md`.

**Review by shadcn — CHANGES_REQUESTED** [review]
@Ventinc This is looking great. Thank you.

I left a few comments.

I'm also seeing a small issue with the display here. See screenshot.

b1bJOhpf

File: apps/www/components/ui/navigation-menu.tsx (hunk 1 above)
Comment (Ventinc) [reply apps/www/components/ui/navigation-menu.tsx:31]: it's used in the NavigationMenuLink to have the same style by default better to remove it ?

File: apps/www/components/ui/navigation-menu.tsx (hunk 3 above)
Comment (Ventinc) [reply apps/www/components/ui/navigation-menu.tsx:54]: I use this variant to change the style of the NavigationMenuLink when it's inside a NavigationMenuContent like ListItem component in demo

File: apps/www/components/ui/navigation-menu.tsx (hunk 2 above)
Comment (Ventinc) [reply apps/www/components/ui/navigation-menu.tsx]: It's already the same displayName no ?
image

**Comment by Ventinc** [comment]
Demo design is fixed too

**Comment by Ventinc** [comment]
I've tried to use the Navigation Menu in an app but there is a problem, https://github.com/radix-ui/primitives/issues/1462

I can fix it by creating a context but it will be too complex to just copy no ? (And put the context in `<MenuNavigation />`)

**Comment by shadcn** [comment]
@Ventinc This is looking good. I have a WIP component for `NavigationMenu` as well. If it's OK with you, I'll push my changes to this PR. We can review here.

**Comment by Ventinc** [comment]
Yeah, let's do this !

**Comment by shadcn** [comment]
@Ventinc I'm updated the code style a bit to match the other components. I've also removed the submenu and vertical demos for now.

**Pull request merged**